core/templates/domain/topic/TopicObjectFactorySpec.ts

```diff
@@ -148,6 +171,31 @@ describe('Topic object factory', () => {
     expect(topic.getAdditionalStoryReferences()).toEqual([]);
     expect(topic.getCanonicalStoryReferences()).toEqual([]);
     expect(topic.getUncategorizedSkillSummaries()).toEqual([]);
+    expect(topic.getDiagnosticTestSkillSummaries()).toEqual([]);
+  });
+
+  it('should be able to correctly update skill Ids for the diagnostic test',
+    () => {
+      var skillSummary1 = ShortSkillSummary.create('skill_1', 'Description 1');
+      var skillSummary2 = ShortSkillSummary.create('skill_1', 'Description 1');
+      expect(_sampleTopic.getDiagnosticTestSkillSummaries()).toEqual(
+        [skillSummary1]);
+      _sampleTopic.setDiagnosticTestSkillSummaries([skillSummary2]);
+      expect(_sampleTopic.getDiagnosticTestSkillSummaries()).toEqual(
+        [skillSummary2]);
+    });
+
+  it('should be able to get all the available skillIds for assignment in the ' +
```

Comment:
Have more tests, include unassigned skills as well, etc.

Try to think of the different scenarios and write tests for each of them. Don't just do the minimum needed to get coverage.

Reply:
Working on tests.

Reply:
OK, noted -- let's keep this thread open until resolution, then.

Reply:
> Have more tests, include unassigned skills as well, etc.

Please help me to understand which test case is missing here because the unassigned skill is included in the test in order to calculate all the available skills( in the last statement).

Reply:
E.g. check what happens if all skill IDs have already been assigned. Or if no skill IDs have been assigned. Or if there are no skills at all.

I'm also a bit confused by the logic here. How does _sampleTopic know about shortSkillSummaries2 and 3? I don't see anything here where we associate those skills with the _sampleTopic (or with anything, really), unless ShortSkillSummary.create() has some side effects which may be problematic.

Reply:
Let, ss = skillSummary
Explanation:
- uncategorizedSkill in the topic: ss1, ss2;
- skill in the subtopic: ss3

Based on the above two points, available skills: ss1, ss2, ss3.

Now diagnostic test contains ss1.
Therefore the available skillSummaries left are ss2 and ss3.

Reply:
Added